src/ui/MessageMenu/messageMenu.tsx

```diff
@@ -0,0 +1,238 @@
+    <MenuItem
+      {...props}
+      onClick={(e) => {
+        if (isUserMessage(message)) copyToClipboard(message?.message);
+        props?.onClick?.(e);
+      }}
+    >
+      {props?.children ?? stringSet.MESSAGE_MENU__COPY}
+    </MenuItem>
+  );
+  const Reply = (props: PrebuildMenuItemPropsType) => (
+    <MenuItem
+      {...props}
+      disabled={message?.parentMessageId > 0}
+      onClick={(e) => {
+        setQuoteMessage?.(message);
+        props?.onClick?.(e);
+      }}
+    >
+      {props?.children ?? stringSet.MESSAGE_MENU__REPLY}
+    </MenuItem>
+  );
+  const Thread = (props: PrebuildMenuItemPropsType) => (
+    <MenuItem
+      {...props}
+      onClick={(e) => {
+        onReplyInThread?.({ message });
+        props?.onClick?.(e);
+      }}
+    >
+      {props?.children ?? stringSet.MESSAGE_MENU__THREAD}
+    </MenuItem>
+  );
+  const OpenInChannel = (props: PrebuildMenuItemPropsType) => (
+    <MenuItem
+      {...props}
+      onClick={(e) => {
+        onMoveToParentMessage?.();
+        props?.onClick?.(e);
+      }}
+    >
+      {props?.children ?? stringSet.MESSAGE_MENU__OPEN_IN_CHANNEL}
+    </MenuItem>
+  );
+  const Edit = (props: PrebuildMenuItemPropsType) => (
+    <MenuItem
+      {...props}
+      onClick={(e) => {
+        if (isOnline) {
+          showEdit?.(true);
+          props?.onClick?.(e);
+        }
+      }}
+    >
+      {props?.children ?? stringSet.MESSAGE_MENU__EDIT}
+    </MenuItem>
+  );
+  const Resend = (props: PrebuildMenuItemPropsType) => (
+    <MenuItem
+      {...props}
+      onClick={(e) => {
+        if (isOnline) {
+          resendMessage?.(message);
+          props?.onClick?.(e);
+        }
+      }}
+    >
+      {props?.children ?? stringSet.MESSAGE_MENU__RESEND}
+    </MenuItem>
+  );
+  const Delete = (props: PrebuildMenuItemPropsType) => (
+    <MenuItem
+      {...props}
+      disabled={
+        typeof disableDeleteMessage === 'boolean'
+          ? disableDeleteMessage
+          : (message?.threadInfo?.replyCount ?? 0) > 0
+      }
+      onClick={(e) => {
+        if (isFailedMessage(message)) {
+          deleteMessage?.(message);
+        } else if (isOnline) {
+          showRemove?.(true);
+          props?.onClick?.(e);
+        }
+      }}
+    >
+      {props?.children ?? stringSet.MESSAGE_MENU__DELETE}
+    </MenuItem>
+  );
+
+  const withAction = (Component, action) => (props: PrebuildMenuItemPropsType) => (
+    <Component {...props} onClick={(e) => { action(); props?.onClick?.(e); }} />
+  );
+  const getPrebuildItems = (handleClose: () => void): PrebuildItemsType => ({
+    Copy: withAction(Copy, handleClose),
+    Reply: withAction(Reply, handleClose),
+    Thread: withAction(Thread, handleClose),
+    OpenInChannel: withAction(OpenInChannel, handleClose),
+    Edit: withAction(Edit, handleClose),
+    Resend: withAction(Resend, handleClose),
+    Delete: withAction(Delete, handleClose),
+  });
```

Comment:
Do not create and use components during rendering. Always define components at the top level.
https://react.dev/learn/your-first-component#nesting-and-organizing-components

Reply:
## Component
```tsx
const Component = () => {
  return <div>component</div>
}

<Component />
```


## Function
```tsx
const renderComponent = () => {
  return <div>component</div>
}

renderComponent();
```